messaging/src/test/java/org/axonframework/eventhandling/processors/streaming/pooled/MergeTaskTest.java

```diff
@@ -49,20 +51,17 @@ class MergeTaskTest {
     private static final int[] SEGMENT_IDS = {0, 1};
     private static final Segment SEGMENT_ZERO = Segment.computeSegment(SEGMENT_TO_MERGE, SEGMENT_IDS);
     private static final Segment SEGMENT_ONE = Segment.computeSegment(SEGMENT_TO_BE_MERGED, SEGMENT_IDS);
-
-    private CompletableFuture<Boolean> result;
     private final Map<Integer, WorkPackage> workPackages = new HashMap<>();
     private final TokenStore tokenStore = mock(TokenStore.class);
-
-    private MergeTask testSubject;
-
     private final WorkPackage workPackageOne = mock(WorkPackage.class);
     private final WorkPackage workPackageTwo = mock(WorkPackage.class);
+    private CompletableFuture<Boolean> result;
+    private MergeTask testSubject;
 
     @BeforeEach
     void setUp() {
         result = new CompletableFuture<>();
-        when(tokenStore.fetchSegments(PROCESSOR_NAME)).thenReturn(SEGMENT_IDS);
+        when(tokenStore.fetchSegments(PROCESSOR_NAME, null)).thenReturn(completedFuture(SEGMENT_IDS));
```

Comment:
When you fix the implementation, then you can use any ProcessingContext in the mock. 
```suggestion
        when(tokenStore.fetchSegments(eq(PROCESSOR_NAME), any(ProcessingContext.class)).thenReturn(completedFuture(SEGMENT_IDS));
```